### PR TITLE
Split deployment checks per network

### LINKED ISSUE #17: Split jobs per network
So that if one network fails because e.g. of a flaky RPC the rest can carry on normally.

See https://github.com/balancer/balancer-deployments/pull/8#pullrequestreview-1424708652.

### PR DESCRIPTION
**Pull request by jubeira**
# Description

This PR splits deployment checks into individual tasks for each network.
This should result in smaller, more localized failures and hopefully faster runs.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- N/A Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

Closes #17.

### CODE REVIEW

File: .github/workflows/deployment-checks.yml [hunk 1]
```diff
@@ -27,19 +27,118 @@ jobs:
...
+      - name: Check Mainnet Deployment Addresses
+        run: yarn check-network-deployments mainnet
+  check-polygon-deployments:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - name: Prepare Config
+        run: yarn ci:prepare-config
```
Comment (ylv-io) [inline .github/workflows/deployment-checks.yml:39]: This code is the same everywhere. Is there a way not to copy paste it?

**Review by ylv-io — APPROVED** [review]
Speed is everything when it comes to development.  Thank you @jubeira .

File: .github/workflows/deployment-checks.yml (hunk 1 above)
Comment (jubeira) [reply .github/workflows/deployment-checks.yml:39]: We can probably do something with composite actions. I'll take a look.

**Comment by ylv-io** [comment]
> > This code is the same everywhere. Is there a way not to copy paste it?
> 
> See [9338a64](https://github.com/balancer/balancer-deployments/commit/9338a64c31d631e96e0973bf7d06445616eb2f59) and [1ad313d](https://github.com/balancer/balancer-deployments/commit/1ad313dd6deecf55f74f1f2c8428eb81f3a86ad5).
> 
> We can isolate the jobs and reuse them, but unfortunately we don't gain _that_ much. Two observations:
> 
> * You still need to do the `checkout` step in the outer workflow _and_ in the inner action.
> * The action cannot use secrets directly (otherwise we could just access all of them at once, and skip the outer `env` step).
>   
>   * Another option would be to send the secrets as inputs, but the problem is that I didn't find a way to define the key inside the action dynamically. In other words, for `mainnet` we need to set `MAINNET_RPC_ENDPOINT` but for `arbitrum` we need to set `ARBITRUM_RPC_ENDPOINT`, so just sending the secret is not enough.
>   * We could also send all the secrets to the actions to skip the outer `env` step, but that doesn't feel correct.

If it is not easy I would not spend time on this. Great work.

**Review by ylv-io — APPROVED** [review]
I meant to approve last time actually.

**Pull request merged**